pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -219,10 +178,14 @@ func TestCreatePod(t *testing.T) {
 	for _, item := range testCases {
 		ctx, cancel := context.WithCancel(context.Background())
 		fakeClientset := fake.NewSimpleClientset()
-		controller := NewNoExecuteTaintManager(ctx, fakeClientset, (&podHolder{pod: item.pod}).getPod, getNodeFromClientset(fakeClientset), getPodsAssignedToNode(fakeClientset))
+		podIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, nil)
+		nodeIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, nil)
+		controller := setupNewNoExecuteTaintManager(ctx, fakeClientset, podIndexer, nodeIndexer)
 		controller.recorder = testutil.NewFakeRecorder()
 		go controller.Run(ctx)
 		controller.taintedNodes = item.taintedNodes
+
+		podIndexer.Add(item.pod)
 		controller.PodUpdated(nil, item.pod)
```

Comment:
I think that you don't need to explicitly call this function if you already added the Pod to the indexer (or the clientset once you do the change I suggested above). The event handler should take care of it.

Reply:
I now use indexers based on the fake clientSet, but still have to call this Add/Update functions. It seems that job_controller_test.go follows this pattern too.